Source/WTF/wtf/URLHelpers.cpp

```diff
@@ -191,7 +203,10 @@ static bool isLookalikeCharacter(const std::optional<UChar32>& previousCodePoint
         || u_hasBinaryProperty(codePoint, UCHAR_DEFAULT_IGNORABLE_CODE_POINT)
         || ublock_getCode(codePoint) == UBLOCK_IPA_EXTENSIONS)
         return true;
-    
+
+    if (isArabicDiacritic(codePoint) && !isArabicCodePoint(previousCodePoint))
+        return true;
```

Comment:
Can't this check be moved to a new specialized template function`isLookalikeSequence<USCRIPT_ARABIC>` and call it in the default section of the following switch statement? In this case, `isArabicDiacritic()` and `isArabicCodePoint()` be moved also as lambda functions inside this new specialized template function.

Reply:
The other ones are a bit different.  They check the code point before and after to make sure they are both the same kind of script.  This only needs to check the code point before.

Reply:
Specifically, if the other code points checked by isLookalikeSequence appear at the beginning of the sequence, it's fine.  If the Arabic diacritics appear at the beginning of the sequence, it is not fine.

Reply:
Oh, this can actually be done.  It's just moving the code to look more like the surrounding code without changing its behavior.